sklearn/mixture/_gaussian_mixture_ic.py

```diff
@@ -756,3 +726,44 @@ def _hierarchical_labels(children, min_components, max_components):
         )
 
     return hierarchical_labels[:, ::-1]
+
+
+class _CollectResults:
+    """
+    Represents the intermediary results for a single GMM clustering run
+    of ``GaussianMixtureIC``
+
+    Attributes
+    ----------
+
+    model : GaussianMixture object
+        GMM clustering fit to the data
+
+    criterion_score : float
+        Bayesian or Aikake Information Criterion score
+
```

Comment:
This is me being a bit pernickety, but `criterion` suffices here as a wording (information criteria are scores).